mkosi/__init__.py

```diff
@@ -4350,17 +4347,24 @@ def install_unified_kernel(
             elif partlabel:
                 boot_options = f"{boot_options} root=PARTLABEL={partlabel}"
 
+            osrelease = root / "usr/lib/os-release"
+            cmdline = workspace(root) / "cmdline"
+            cmdline.write_text(boot_options)
+            initrd = root / prefix / args.machine_id / kver / "initrd"
+
             cmd: Sequence[PathString] = [
-                "dracut",
-                "--uefi",
-                "--kver", kver,
-                "--kernel-image", Path("/") / kimg,
-                "--kernel-cmdline", boot_options,
-                "--force",
+                "objcopy",
+                "--add-section", f".osrel={osrelease}",   "--change-section-vma", ".osrel=0x20000",
+                "--add-section", f".cmdline={cmdline}",   "--change-section-vma", ".cmdline=0x30000",
+                "--add-section", f".linux={root / kimg}", "--change-section-vma", ".linux=0x2000000",
+                "--add-section", f".initrd={initrd}",     "--change-section-vma", ".initrd=0x3000000",
```

Comment:
I'll repeat the comment from the other PR, so this doesn't get lost:
dracut varied this by arch: 0x3000000 by default, but 0x4000000 on arm64. Was there some discussion of this value?

Reply:
I think I just stole this from dracut when I initially wrote this, but I might have gotten it wrong, will double check again.